src/mechanics/input.rs

```diff
@@ -137,12 +172,85 @@ pub fn move_player(
             )
             .is_some()
             {
-                player_movement_broadcast.send(PlayerMovementActions::Bumping);
+                entity_movement_broadcast.send(PlayerMovementActions::Bumping);
+                commands.entity(entity).remove::<MovementIntent>();
                 return;
             }
         }
-        player_transform.translation = projected_position;
-        player_movement_broadcast.send(PlayerMovementActions::Walking);
+
+        entity_transform.translation = projected_position;
+        entity_movement_broadcast.send(PlayerMovementActions::Walking);
+        commands.entity(entity).remove::<MovementIntent>();
+    }
+}
+
+pub fn interact_entity(
+    input: Res<Input<KeyCode>>, 
+    tile_query: Query<&EntityInstance>,
+    entity_query: Query<(&Transform, &DirectionFacing)>,
+    level_dimension: Res<LevelDimensions>,
+)
+{
+    if entity_query.is_empty() {
+        return;
+    }
+
+    if input.just_pressed(KeyCode::E) {
+        let interactive_tiles = tile_query
+            .iter()
+            .filter(|&tile| !tile.field_instances.is_empty())
+            .filter(|&tile| {
+                tile.field_instances
+                    .iter()
+                    .any(|field_instance| field_instance.identifier == "Interactable")
+            })
+            .collect::<Vec<&EntityInstance>>();
+
+        let (entity_transform, facing) = entity_query.get_single().unwrap();
+        
+        let pixel_distance = 3.0;
+        let mut direction = Vec3::ZERO;
+
+        match facing {
+            DirectionFacing::Up => {
+                direction += Vec3::new(0.0, pixel_distance, 0.0);
+            }
+            DirectionFacing::Down => {
+                direction -= Vec3::new(0.0, pixel_distance, 0.0);
+            }
+            DirectionFacing::Left => {
+                direction -= Vec3::new(pixel_distance, 0.0, 0.0);
+            }
+            DirectionFacing::Right => {
+                direction += Vec3::new(pixel_distance, 0.0, 0.0);
+            }
+        }
+
+        let tile_side_length = 64.0;
+        let projected_position = entity_transform.translation + direction;
+
+        for &interactive_tile in interactive_tiles.iter() {
+            let tile_position = Vec3::new(
+                interactive_tile.px.x as f32,
+                (level_dimension.height as i32 - (interactive_tile.px.y)) as f32,
+                0.0,
+            );
+
+            if collide(
+                projected_position,
+                Vec2::new(tile_side_length, tile_side_length),
+                tile_position,
+                Vec2::new(interactive_tile.width as f32, interactive_tile.height as f32),
+            )
+            .is_some()
+            {
+                let text = interactive_tile.field_instances().get(1).unwrap();
```

Comment:
I'd recommend replacing .unwrap() statements with .expect("Some helpful error message here.") just to give more meaning to crashes. For instance, I typically craft expect statement messages to give me information on
- Where the crash occurred.
- Which assumptions were violated.

So for this line, for example, I would craft it like this:
```rust
let text = interactive_tile.field_instances().get(1)
    .expect("interact_entity: Could not find Interactive text in Interactive Tile.");
```

Reply:
LDtk has an option to disallow the text to be null but I agree that using an expect would be good practice